src/nano_contracts/deserializer.ts

```diff
@@ -75,14 +72,17 @@ class Deserializer {
   deserializeContainerType(
     buf: Buffer,
     type: string
-  ): DeserializeResult<NanoContractArgumentType | null> {
+  ): BufferROExtract<NanoContractArgumentType | null> {
     const [containerType, internalType] = getContainerInternalType(type);
 
     switch (containerType) {
       case 'Optional':
         return this.toOptional(buf, internalType);
+      case 'RawSignedData':
       case 'SignedData':
```

Comment:
Both serialize to the same bytes.
The only difference is that the signature of `SignedData` also includes the contract id, which is not part of the serialized field.